testutils/clustermanager/gke.go

```diff
@@ -280,6 +199,10 @@ func (gc *GKECluster) Acquire() error {
 		if err != nil {
 			return fmt.Errorf("failed getting cluster name: '%v'", err)
 		}
+		gc.Request.ClusterName = clusterName
+	}
+	if gc.Request.Project == "" {
+		gc.Request.Project = *gc.Project
```

Comment:
As we discussed earlier, we should keep the Request immutable

Reply:
Instead we should make a deep copy of `Request`, and modify that one instead

Reply:
Done. Added the deepcopy logic.